src/main/java/hudson/plugins/s3/S3ArtifactsAction.java

```diff
@@ -66,6 +66,10 @@ public String getProfile() {
         return artifacts;
     }
 
+    public void addArtifacts(List<FingerprintRecord> artifacts) {
```

Comment:
This method is not needed. I'm not a Jenkins expert, but usually I'm trying to avoid creation of new public methods :)

You can use `getArtifacts().addAll()`. It should work as well.